File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        # for i in range(len(nums)):                   #n
        #     need=target-nums[i]
        #     if need in nums and nums.index(need)!=i: #n
        #         return [i,nums.index(need)]          #total n^2

        
        
        #use dictonary to check O(1)
        d={}
        for i in range(len(nums)):
            d[nums[i]]=i
        for i in range(len(nums)):
            need = target - nums[i]
            if need in d.keys() and d[need]!=i:
                return [i,d[need]]


        

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int i=0;
       
        while(i<n-1){
            int j=i+1;
            while(j<n){
                if(nums[i]+nums[j]==target){
                    return new int[]{i, j};
                }
                else{
                    j++;
                }
            }
            i++;
        }
        return new int[]{};
    }
}

